service/lib/dinstaller/manager.rb

```diff
@@ -167,54 +170,11 @@ def probe_languages
       self.language = DEFAULT_LANGUAGE
     end
 
+    # Probes storage devices and performs an initial proposal
     def probe_storage
-      logger.info "Probing storage"
-      storage_manager.probe
-      @disks = storage_manager.probed.disks
-      self.disk = @disks.first&.name
-    end
-
-    # @return [Boolean] true if success; false if failed
-    def propose_storage(disk_name)
-      settings = Y2Storage::ProposalSettings.new_for_current_product
-      settings.candidate_devices = [disk_name]
-
-      # FIXME: clean up the disks
-      clean_probed = storage_probed.clone
-      clean_probed.disks.each(&:remove_descendants)
-
-      proposal = Y2Storage::GuidedProposal.initial(
-        devicegraph: clean_probed,
-        settings:    settings
-      )
-      return false if proposal.failed?
-
-      storage_manager.proposal = proposal
-      true
-    end
-
-    def storage_probed
-      storage_manager.probed
-    end
-
-    def storage_manager
-      @storage_manager ||= Y2Storage::StorageManager.instance
-    end
-
-    def initialize
-      Yast::Mode.SetUI("commandline")
-      Yast::Mode.SetMode("installation")
-      @disks = []
-      @languages = []
-      @products = []
-      @status_callbacks = []
-      @status = InstallerStatus::ERROR # it should start with probing, so just temporary status
-      @logger = logger || Logger.new($stdout)
-      @progress = Progress.new
-      @software = Software.new(@logger)
-      # Set stage to initial, so it will act as installer for some cases like
-      # proposing installer instead of reading current one
-      Yast::Stage.Set("initial")
+      logger.info "Probing storage and performing proposal"
+      Y2Storage::StorageManager.instance.probe
```

Comment:
is possible to get some progress from that probing?

Reply:
Well, potentially libstorage-ng can trigger a callback for every step of the process. But it's not exactly progress reporting (nothing like a percentage or so), it's more a live log of the actions libstorage-ng is performing. See https://github.com/yast/yast-storage-ng/blob/master/src/lib/y2storage/callbacks/probe.rb#L57

Reply:
@ancorgs well, progress also does not have percentages, but it has substeps there. So if you know number of steps, then reporting is fine. If not, then we can add ability to progress API to have like living substeps that just change message and does not do counting of substeps.